contracts/staking/stakeManager/StakeManager.sol

```diff
@@ -422,44 +437,63 @@ contract StakeManager is
         );
 
         uint256 exitEpoch = currentEpoch.add(1); // notice period
-        _unstake(validatorId, exitEpoch);
+        _unstake(validatorId, exitEpoch, pol);
     }
 
-    function transferFunds(
-        uint256 validatorId,
-        uint256 amount,
-        address delegator
-    ) external returns (bool) {
-        require(
-            validators[validatorId].contractAddress == msg.sender ||
-                Registry(registry).getSlashingManagerAddress() == msg.sender,
-            "not allowed"
-        );
-        return token.transfer(delegator, amount);
+    function transferFunds(uint256 validatorId, uint256 amount, address delegator) external returns (bool) {
+        return _transferFunds(validatorId, amount, delegator, false);
     }
 
-    function delegationDeposit(
-        uint256 validatorId,
-        uint256 amount,
-        address delegator
-    ) external onlyDelegation(validatorId) returns (bool) {
-        return token.transferFrom(delegator, address(this), amount);
+    function transferFundsPOL(uint256 validatorId, uint256 amount, address delegator) external returns (bool) {
+        return _transferFunds(validatorId, amount, delegator, true);
     }
 
-    function stakeFor(
-        address user,
-        uint256 amount,
-        uint256 heimdallFee,
-        bool acceptDelegation,
-        bytes memory signerPubkey
-    ) public onlyWhenUnlocked {
+    function _transferFunds(uint256 validatorId, uint256 amount, address delegator, bool pol) internal returns (bool) {
+        require(validators[validatorId].contractAddress == msg.sender || Registry(registry).getSlashingManagerAddress() == msg.sender, "not allowed");
+        if (!pol) _convertPOLToMatic(amount);
+        IERC20 token_ = _getToken(pol);
+        return token_.transfer(delegator, amount);
+    }
+
+    function delegationDeposit(uint256 validatorId, uint256 amount, address delegator) external onlyDelegation(validatorId) returns (bool) {
+        return _delegationDeposit(amount, delegator, false);
+    }
+
+    function delegationDepositPOL(uint256 validatorId, uint256 amount, address delegator) external onlyDelegation(validatorId) returns (bool) {
+        return _delegationDeposit(amount, delegator, true);
+    }
+
+    function _delegationDeposit(uint256 amount, address delegator, bool pol) internal returns (bool) {
+        IERC20 token_ = _getToken(pol);
+        bool result = token_.transferFrom(delegator, address(this), amount);
```

Comment:
Can refactor this -

`_transferTokenFrom(delegator, address(this), amount, pol)`